### PR TITLE
Checkbox

### PR DESCRIPTION
**Pull request by varl**
- **status** attr not implemented, waiting on specs from @joeDHIS

### CODE REVIEW

File: src/core/Checkbox/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        indeterminate: !!this.props.indeterminate,
+    }
+
+    onClick = () => {
+        const checked = !this.state.checked
+        this.props.onChange(this.props.name, checked)
+        this.setState({ checked, indeterminate: false })
+    }
+
+    icons = {
```
Comment (adeelshahid) [inline src/core/Checkbox/index.js]: we can make them static or move them out of `class Checkbox`

File: src/core/Checkbox/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                    })}
+                >
+                    {this.props.label}
+                </span>
+            </label>
+        )
+    }
+}
+
+Checkbox.defaultProps = {
```
Comment (adeelshahid) [inline src/core/Checkbox/index.js]: default `required: false`

File: docs/src/demos/Atoms/Input/Checkbox/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                    name="indeterminate-checkbox"
+                    label="Indeterminate Checkbox"
+                    onChange={(name, v) => this.onChange(name, v)}
+                />
+                <Checkbox
+                    disabled
+                    name="disabled-checkbox"
+                    label="Disabled Checkbox"
+                    onChange={(name, v) => this.onChange(name, v)}
+                />
```
Comment (adeelshahid) [inline docs/src/demos/Atoms/Input/Checkbox/index.js]: - `<Divider />` before `</div>`
- Demo for `['valid', 'error', 'warning']` states

File: src/core/Checkbox/index.js (hunk 2 above)
Comment (varl) [reply src/core/Checkbox/index.js]: 71b5ad8

File: src/core/Checkbox/index.js (hunk 1 above)
Comment (varl) [reply src/core/Checkbox/index.js]: 71b5ad8

File: docs/src/demos/Atoms/Input/Checkbox/index.js (hunk 3 above)
Comment (varl) [reply docs/src/demos/Atoms/Input/Checkbox/index.js]: Divider: 71b5ad8

will fix demo for various states once i talk to joe

**Pull request merged**